internal/csi-addons/networkfence/fencing.go

```diff
@@ -287,6 +287,57 @@ func (nf *NetworkFence) AddClientEviction(ctx context.Context) error {
 	return nil
 }
 
+func getBlocklistEntries(ctx context.Context) ([]string, error) {
+	// FIXME: replace the ceph command with go-ceph API in future
+	cmd := []string{"osd", "blocklist", "ls"}
+	stdout, stdErr, err := util.ExecCommandWithTimeout(ctx, 5*time.Minute, "ceph", cmd...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list blocklisted clients: %w, stderr: %q", err, stdErr)
+	}
+
+	// Parse the stdout to extract blocklist entries
+	// each entry is a separate line in the output
+	entries := strings.Split(strings.TrimSpace(stdout), "\n")
+
+	return entries, nil
+}
+
+func removeBlocklistEntry(ctx context.Context, entry string) error {
+	// FIXME: replace the ceph command with go-ceph API in future
+	cmd := []string{"osd", "blocklist", "rm", entry}
+	_, stdErr, err := util.ExecCommandWithTimeout(ctx, 5*time.Minute, "ceph", cmd...)
+	if err != nil {
+		return fmt.Errorf("failed to un-blocklist entry %s: %w, stderr: %q", entry, err, stdErr)
+	}
+	log.DebugLog(ctx, "un-blocklisted entry %s successfully", entry)
+
+	return nil
+}
+
+// RemoveClientEviction unblocks access for all the IPs in the IP range mentioned via the CIDR block
+// using a network fence.
+func (nf *NetworkFence) RemoveClientEviction(ctx context.Context) error {
+	// get the list of blocklist entries
+	blocklistEntries, err := getBlocklistEntries(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get blocklist entries: %w", err)
+	}
+
+	// iterate through CIDR blocks and remove matching entries
+	for _, cidr := range nf.Cidr {
+		for _, entry := range blocklistEntries {
+			if strings.HasPrefix(entry, cidr) {
```

Comment:
It is difficult to verify how this works. Make it easier for others (and for yourself in a few months time) to understand this by adding comments with examples of what `cidr` contains, and how `enty` is formatted.

Reply:
Thanks @nixpanic , updated. PTAL

Reply:
A `cidr` in the format of `127.0.0.1:0/3710147553` is weird...

A CIDR could be `192.168.123.234/16`, a blocklisted IP in that range can be `192.168.1.2` (note that the prefix-comparing won't work here). You will need to check if the IP-address from the `entry` is included in the range of IP-addresses described by the CIDR. At the moment, the current check is not correct.

Reply:
Thanks @nixpanic ,
Updated.